src/Types/RawBytes.purs

```diff
@@ -0,0 +1,79 @@
+-- | A wrapper over `ByteArray` to indicate a byte array with no further specified meaning
+
+module Types.RawBytes
+  ( RawBytes(..)
+  , rawBytesToByteArray
+  , rawBytesFromByteArray
+  , rawBytesToIntArray
+  , rawBytesFromIntArray
+  , rawBytesFromIntArrayUnsafe
+  , rawBytesToHex
+  , byteLength
+  , hexToRawBytes
+  , hexToRawBytesUnsafe
+  ) where
+
+import Data.Newtype (class Newtype, wrap, unwrap)
+
+import Types.ByteArray (ByteArray)
+import Types.ByteArray as BA
+import Data.Maybe (Maybe)
+import Prelude
+import Aeson (class DecodeAeson, decodeAesonViaJson)
+import Data.Argonaut (class DecodeJson, decodeJson)
+import Test.QuickCheck.Arbitrary (class Arbitrary, arbitrary)
+
+-- | An array of Bytes with no information about the content format
+newtype RawBytes = RawBytes ByteArray
+
+instance Show RawBytes where
+  show rb = "(rawBytesFromIntArrayUnsafe" <> show (rawBytesToIntArray rb) <> ")"
+
+derive instance Newtype RawBytes _
+
+derive instance Eq RawBytes
+
+derive instance Ord RawBytes
+
+instance Semigroup RawBytes where
+  append r1 r2 = wrap $ unwrap r1 `append` unwrap r2
+
+instance Monoid RawBytes where
+  mempty = rawBytesFromIntArrayUnsafe []
+
+instance DecodeJson RawBytes where
+  decodeJson = map wrap <<< decodeJson
+
+instance DecodeAeson RawBytes where
+  decodeAeson = decodeAesonViaJson
+
+instance Arbitrary RawBytes where
+  arbitrary = wrap <$> arbitrary
```

Comment:
Again, I'd newtype derive where possible
```

derive newtype instance Eq RawBytes
derive newtype instance Ord RawBytes
derive newtype instance Semigroup RawBytes
derive newtype instance Monoid RawBytes
derive newtype instance DecodeJson RawBytes
derive newtype instance DecodeAeson RawBytes
derive newtype instance Arbitrary RawBytes
```

You'll need to clean up your redundant imports after this.